src/object/srv_obj.c

```diff
@@ -709,9 +1020,24 @@ ds_obj_rw_local_hdlr(crt_rpc_t *rpc, uint32_t tag, struct ds_cont_hdl *cont_hdl,
 	err = bio_iod_post(biod);
 	rc = rc ? : err;
 out:
+	rc = ds_obj_rw_complete(rpc, cont_hdl, *ioh, rc, map_version, dth,
+				dti_cos_count, dti_cos);
 	return rc;
 }
 
+static int
+ds_update_sanity_check(daos_handle_t coh, struct obj_rw_in *orw)
+{
+	/*
+	 * TBD: POSIX semantics check, including:
+	 *	1. Permission check.
+	 *	2. Cannot insert the same name entry (dkey) repeatedly.
+	 *	3. Cannot update akey if the related name entry does not exist.
+	 */
+
+	return 0;
+}
```

Comment:
Here, the sanity check mainly used for some special semantics check, such as POSIX, that is only required conditionally. For example, only leader to do that for non-resend case. So I would suggest to keep the check_container and pool_map version check outside.

Reply:
what is the difference? these are all accessment check. Btw: if it is always 0, I would suggest to remove them, in case it would cause some wrong assumption?

Reply:
This part is not related with DTX work, I put it there just for the TBD list. I can remove it.